### PR TITLE
Add coqPackages.mathcomp-apery

### PR DESCRIPTION
**Pull request by proux01**
###### Description of changes

https://github.com/coq-community/apery

Useful for mathcomp's CI

Successfully tested on https://github.com/coq-community/coq-nix-toolbox/pull/139

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by proux01** [comment]
@vbgl successfully tested in coq-nix-toolbox: https://github.com/coq-community/coq-nix-toolbox/pull/139
We'd like to have Apery for the CI of mathcomp

File: pkgs/top-level/coq-packages.nix [hunk 1]
```diff
@@ -85,6 +85,7 @@ let
       mathcomp-zify = callPackage ../development/coq-modules/mathcomp-zify {};
       mathcomp-algebra-tactics = callPackage ../development/coq-modules/mathcomp-algebra-tactics {};
       mathcomp-tarjan = callPackage ../development/coq-modules/mathcomp-tarjan {};
+      mathcomp-apery = callPackage ../development/coq-modules/mathcomp-apery {};
```
Comment (vbgl) [inline pkgs/top-level/coq-packages.nix:88]: Isn’t this list supposed to be in alphabetical order?

Comment (vbgl) [reply pkgs/top-level/coq-packages.nix]: Well done. Thanks.

**Review by vbgl — APPROVED** [review]

**Comment by proux01** [comment]
Thanks!

**Pull request merged**